src/instructlab/sdg/datamixing.py

```diff
@@ -94,29 +89,67 @@ class Recipe:
     """
 
     def __init__(
-        self, initial_datasets: Optional[list] = None, sys_prompt: Optional[str] = ""
+        self, recipe_path: Optional[str] = None, sys_prompt: Optional[str] = ""
     ):
-        self.recipe = {
-            "datasets": initial_datasets or [],
-            "sys_prompt": sys_prompt,
-        }
-        self.sys_prompt = self.recipe.get("sys_prompt", "")
+        self.recipe_path = recipe_path or ""
+        self.sys_prompt = sys_prompt
+
+        # Defaults if no recipe path given or these values don't
+        # exist in the given recipe file
+        self.datasets = []
+        if recipe_path is not None:
+            recipe = self._load_recipe()
+            if "datasets" in recipe:
+                self.datasets = recipe["datasets"]
+
         self.dataset_added = False
 
+    def _load_recipe(self):
+        with open(self.recipe_path, encoding="utf-8") as fp:
+            return yaml.safe_load(fp)
+
+    def _load_ds(self, path):
+        """
+        Load a dataset from the given location. If a jsonl file is
+        given, we load the dataset from disk. Otherwise, we load the
+        path given from HuggingFace. Relative paths are resolved
+        respective to the directory the recipe yaml itself resides in.
+        """
+        if path.endswith(".jsonl"):
+            if not os.path.isabs(path):
+                path = os.path.join(os.path.dirname(self.recipe_path), path)
+            logger.info(f"Loading dataset from {path} ...")
+            dataset = load_dataset("json", data_files=path, split="train")
+        else:
+            logger.info(f"Loading dataset from HF {path} ...")
+            dataset = load_dataset(path, split="train")
```

Comment:
I thought we were removing this?

Reply:
Hugging Face download mentioned in docstring too

Reply:
Yes, we are! However in my multi-tasking it looks like I accidentally did that in a different local branch than the one I pushed here. Need more coffee, will fix as I have a rebase conflict to fix anyway.

Reply:
Sorry, fixed that now.